### PR TITLE
Revert "Revert "Use showSupportMessaging flag from members-data-api""

### PR DESCRIPTION
**Pull request by tjsilver**
Re-implements guardian/frontend#21290 but with the addition of other product-related cookies temporarily. This needs to be in place for at least 24 hours and then we'll do a clean-up PR.

Reverts guardian/frontend#21305

### CODE REVIEW

File: common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js [hunk 1]
```diff
@@ -88,8 +88,15 @@
...
 
-    function isPayingMember() {
-        return getCookieValue('gu_paying_member') === 'true';
+
+    function shouldHideSupportMessaging() {
+        return getCookieValue('gu_paying_member') === 'true' ||
+        getCookieValue('gu_digital_subscriber') === 'true' ||
+        getCookieValue('gu_recurring_contributor') === 'true' ||
+        (
+            !!getCookieValue('gu_show_support_messaging') &&
```
Comment (paulbrown1982) [inline common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js]: Is this testing for ‘is not null’? If so, then that’s covered by the triple equals on the next line after the &&. If alternatively it’s testing for ‘is Boolean false’ then the && should be ||

File: common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js [hunk 2]
```diff
@@ -140,8 +147,8 @@
         documentElement.style.fontSize = baseFontSize
     }
 
-    if (isPayingMember()) {
-        docClass += ' is-paying-member';
+    if (shouldHideSupportMessaging()) {
+        docClass += ' hide-support-messaging'
```
Comment (paulbrown1982) [inline common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js]: There’s a missing semicolon here.

**Review by paulbrown1982 — CHANGES_REQUESTED** [review]

File: common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js (hunk 2 above)
Comment (tjsilver) [reply common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js]: Thanks - will add.

File: common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js (hunk 1 above)
Comment (tjsilver) [reply common/app/templates/inlineJS/blocking/applyRenderConditions.scala.js]: That was a one-line replacement of the previous code: 
`var value = getCookieValue('gu_show_support_messaging');
        if (!value) {
            return false;
        }
        return value === 'false';`
However, it isn't very readable so I'll instead change the cookie to 'hide support messaging' to make it easier to reason about.

**Review by twrichards — APPROVED** [review]
Awesome stuff 👍
Much simpler now the cookie is 'hide' rather than 'show' - makes everything easier - was really worth your thorough thinking it through.

**Comment by tjsilver** [comment]
@rupertbates Have created a new cookie called 'gu_hide_support_messaging'

**Review by paulbrown1982 — APPROVED** [review]
👍 Looks great!

**Pull request merged**